_posts/2024-02-01-wildfly-glow.adoc

```diff
@@ -0,0 +1,504 @@
+present in the aggegator layer (eg: `cloud-server` or `jaxrs-server`) then the gain is lower.
+
+My colleague Kabir Khan has wrote an interesting https://github.com/kabir/vlog-glow/blob/main/README.md[project] 
+and associated Vlog that showcases the gain you can expect with WildFly Glow. If you are interested by WildFly Glow 
+you should really follow his Vlog, it is a very nice video on the topic.
+
+== Datasources support, missing datasource detection
+
+WildFly Glow detects that your deployment uses datasources and will abort asking you to take an action to fix the problem.
+It will suggest the set of known add-ons allowing to connect to database.
+
+You will have to choose one of the proposed add-ons. 
+
+An example of reported error when using the WildFly Glow CLI:
+
+[source,bash]
+----
+wildfly-glow scan todo-backend.war
+
+Wildfly Glow is scanning...
+context: bare-metal
+enabled profile: none
+galleon discovery
+- feature-packs
+   org.wildfly:wildfly-galleon-pack:30.0.1.Final
+- layers
+   ee-core-profile-server
+   ejb-lite
+   jpa
+   jaxrs
+
+identified errors
+* unbound datasources error: java:jboss/datasources/ToDos
+  To correct this error, enable one of the following add-ons:
+  - mariadb
+  - mssqlserver
+  - mysql
+  - oracle
+  - postgresql
+
+Some suggestions have been found. You could enable suggestions with --suggest option (if using the WildFly Glow CLI) or <suggest>true</suggest> (if using the WildFly Maven Plugin).
+To enable add-ons add the --add-ons=<list of add-ons> option to the scan command
+Some errors have been reported, you should fix them prior to provision a server with the --provision=SERVER option of the scan command
+----
+
+== Datasources support, setting a datasource add-on
+
+We are using the `postgresql` add-on to fix the reported error.
+
+[source,bash]
+----
+wildfly-glow scan todo-backend.war --add-ons=postgresql
+
+Wildfly Glow is scanning...
+context: bare-metal
+enabled profile: none
+galleon discovery
+- feature-packs
+   org.wildfly:wildfly-galleon-pack:30.0.1.Final
+   org.wildfly:wildfly-datasources-galleon-pack:6.0.0.Final
+- layers
+   ee-core-profile-server
+   ejb-lite
+   jpa
+   jaxrs
+   postgresql-datasource
+
+enabled add-ons
+- postgresql : Documentation in https://github.com/wildfly-extras/wildfly-datasources-galleon-pack
+
+strongly suggested configuration
+
+postgresql-datasource:
+ - POSTGRESQL_DATABASE=Defines the database name to be used in the datasource’s `connection-url` property.
+ - POSTGRESQL_JNDI=java:jboss/datasources/ToDos
+ - POSTGRESQL_PASSWORD=Defines the password for the datasource.
+ - POSTGRESQL_USER=Defines the username for the datasource.
+
+----
+
+WildFly Glow detects that the postgresql datasource has to be configured with env variables and advertise the set of environment variables to be used when starting the server.
+
+Example of a server started with env variables set:
+
+[source,bash]
+----
+POSTGRESQL_DATABASE=test \
```

Comment:
I'm a bit wooly, but do we need to export these to make them available to the script?

Reply:
Yes, we need to set them at server startup.